x/data/server/lookup/table_test.go

```diff
@@ -0,0 +1,256 @@
+package lookup
+
+import (
+	"hash"
+	"hash/fnv"
+	"math"
+	"testing"
+
+	"github.com/cosmos/cosmos-sdk/store/mem"
+	"github.com/stretchr/testify/require"
+
+	"github.com/tendermint/tendermint/libs/rand"
+)
+
+func TestTable(t *testing.T) {
+	// test default case with good params
+	table, err := NewTable(nil)
+	require.NoError(t, err)
+	testTable(t, table, 5)
+
+	// test suboptimal case to trigger varint fallback
+	table, err = NewTableWithOptions(TableOptions{
+		MinLength: 1,
+		NewHash: func() hash.Hash {
+			return sixteenByteHash{
+				fnv.New32(),
+			}
+		},
+	})
+	require.NoError(t, err)
+	testTable(t, table, 5)
+}
+
+type sixteenByteHash struct {
+	hash.Hash
+}
+
+func (h sixteenByteHash) Sum(b []byte) []byte {
+	bz := h.Hash.Sum(b)
+	// just return two bytes
+	return bz[:len(b)+2]
```

Comment:
```suggestion
	return bz[:2]
```

Reply:
otherwise we should update the comment.

Reply:
I'll update the comment. This is to conform to the specs of `Sum`